aggregator/fetchers/fetchers_test.go

```diff
@@ -0,0 +1,211 @@
+			}
+			require.Equal(t, errInvalidResponseData, err)
+			require.Equal(t, float64(0), price)
+		})
+		t.Run("invalid string for price should error"+fetcherName, func(t *testing.T) {
+			t.Parallel()
+
+			fetcher, _ := NewPriceFetcher(fetcherName, &mock.HttpResponseGetterStub{
+				GetCalled: getFuncGetCalled(fetcherName, "not a number", pair, nil),
+			})
+			assert.False(t, check.IfNil(fetcher))
+
+			price, err := fetcher.FetchPrice(context.Background(), ethTicker, quoteUSDFiat)
+			if err == errShouldSkipTest {
+				return
+			}
+			require.NotNil(t, err)
+			require.Equal(t, float64(0), price)
+			require.IsType(t, err, &strconv.NumError{})
+		})
+		t.Run("should work eth-usd"+fetcherName, func(t *testing.T) {
+			t.Parallel()
+
+			fetcher, _ := NewPriceFetcher(fetcherName, &mock.HttpResponseGetterStub{
+				GetCalled: getFuncGetCalled(fetcherName, "4714.05000000", pair, nil),
+			})
+			assert.False(t, check.IfNil(fetcher))
+
+			price, err := fetcher.FetchPrice(context.Background(), ethTicker, quoteUSDFiat)
+			if err == errShouldSkipTest {
+				return
+			}
+			require.Nil(t, err)
+			require.Equal(t, 4714.05, price)
+			assert.Equal(t, fetcherName, fetcher.Name())
+		})
+		t.Run("should work btc-usd"+fetcherName, func(t *testing.T) {
+			t.Parallel()
+
+			btcTicker := "BTC"
+			btcUsdPair := btcTicker + quoteUSDFiat
+			fetcher, _ := NewPriceFetcher(fetcherName, &mock.HttpResponseGetterStub{
+				GetCalled: getFuncGetCalled(fetcherName, "4714.05000000", btcUsdPair, nil),
+			})
+			assert.False(t, check.IfNil(fetcher))
+
+			price, err := fetcher.FetchPrice(context.Background(), btcTicker, quoteUSDFiat)
+			if err == errShouldSkipTest {
+				return
+			}
+			require.Nil(t, err)
+			require.Equal(t, 4714.05, price)
+			assert.Equal(t, fetcherName, fetcher.Name())
+		})
+	}
+}
+
+func getFuncGetCalled(name, returnPrice, pair string, returnErr error) func(ctx context.Context, url string, response interface{}) error {
+	switch name {
+	case binanceName:
+		return func(ctx context.Context, url string, response interface{}) error {
+			cast, _ := response.(*binancePriceRequest)
+			cast.Price = returnPrice
+			return returnErr
+		}
+	case bitfinexName:
+		return func(ctx context.Context, url string, response interface{}) error {
+			cast, _ := response.(*bitfinexPriceRequest)
+			cast.Price = returnPrice
+			return returnErr
+		}
+	case cryptocomName:
+		return func(ctx context.Context, url string, response interface{}) error {
+			cast, _ := response.(*cryptocomPriceRequest)
+			var err error
+			cast.Result.Data.Price, err = strconv.ParseFloat(returnPrice, 64)
+			if err != nil {
+				return errShouldSkipTest
+			}
+			return returnErr
+		}
+	case geminiName:
+		return func(ctx context.Context, url string, response interface{}) error {
+			cast, _ := response.(*geminiPriceRequest)
+			cast.Price = returnPrice
+			return returnErr
+		}
+	case hitbtcName:
+		return func(ctx context.Context, url string, response interface{}) error {
```

Comment:
this is function definition is duplicated multiple times. Maybe just initialize a function that take a function and returning just this at the end of the current function.

Reply:
as discussed, we'll keep it as it is until we find a way to properly remove duplicated code